pallets/order-book/src/lib.rs

```diff
@@ -509,10 +512,14 @@ pub mod pallet {
             let order_book =
                 <OrderBooks<T>>::get(order_book_id).ok_or(Error::<T>::UnknownOrderBook)?;
 
-            let mut data = CacheDataLayer::<T>::new();
-            let count_of_canceled_orders = order_book.cancel_all_limit_orders(&mut data)? as u32;
+            ensure!(
+                order_book.status == OrderBookStatus::OnlyCancel
+                    || order_book.status == OrderBookStatus::Stop,
+                Error::<T>::ForbiddenStatusToDeleteOrderBook
+            );
 
-            data.commit();
+            let count = <LimitOrders<T>>::iter_prefix_values(order_book_id).count();
```

Comment:
```suggestion
            let is_empty = <LimitOrders<T>>::iter_prefix_values(order_book_id).next().is_none();
```
We don't need to iterate over all orders